client.go

```diff
@@ -68,47 +78,145 @@ func getDockerClient() DockerClient {
 	return dockerClient
 }
 
-// Run the given command in a container based on the given image. The given
-// image string is just the ID of said image. It returns true if the command
-// was successful, false otherwise.
-func runCommandInContainer(img string, cmd []string) bool {
-	client := getDockerClient()
+// Looks for the specified command inside of a Docker image.
+// The given image string is just the ID of said image.
+// It returns true if the command was successful, false otherwise.
+func checkCommandInImage(img string, cmd string) bool {
+	wait := func(id string) error {
+		client := getDockerClient()
+
+		select {
+		case res := <-client.Wait(id):
+			if res.Error != nil {
+				return res.Error
+			} else if res.ExitCode != 0 {
+				return fmt.Errorf("Command exited with %i\n", res.Error)
+			}
+		case <-time.After(containerTimeout):
+			return fmt.Errorf("Timed out when waiting for a container.\n")
+		}
+		return nil
+	}
 
-	// First of all we create a container in which we will run the command.
-	config := &dockerclient.ContainerConfig{Image: img, Entrypoint: cmd}
-	id, err := client.CreateContainer(config, "")
-	if err != nil {
+	containerId, err := startContainer(img, []string{cmd}, false, wait)
+
+	defer removeContainer(containerId)
+
+	if err == nil {
+		return true
+	} else {
 		log.Println(err)
 		return false
 	}
-	defer removeContainer(client, id)
+}
+
+// Run the given command in a container based on the given image. The given
+// image string is just the ID of said image.
+// The STDOUT and STDERR of the container can be streamed to the host's STDOUT
+// by setting the `streaming` parameter to true.
+// It returns the ID of the container spawned from the image.
+// Note well: the container is NOT deleted when the given command terminates.
+func runCommandInContainer(img string, cmd []string, streaming bool) (string, error) {
+	wait := func(id string) error {
+		client := getDockerClient()
+		select {
+		case res := <-client.Wait(id):
+			if res.Error != nil {
+				return res.Error
+			} else {
```

Comment:
else -> else if. But as I said in a comment, maybe you can factor this code out in a function `handleWait` or something like that.